src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -2709,15 +2619,17 @@ BuildRoutesForInsert(Query *query, DeferredErrorMessage **planningError)
 	{
 		InsertValues *insertValues = (InsertValues *) lfirst(insertValuesCell);
 		List *prunedShardIntervalList = NIL;
+		Expr *partitionValueExpr = (Expr *) strip_implicit_coercions(
+			(Node *) insertValues->partitionValueExpr);
```

Comment:
This is to make the function consistent with ExtractInsertPartitionKeyValue, such that we can determine the value here if ExtractInsertPartitionKeyValue returns value.